resources/js/components/shared/TimelinePlot.vue

```diff
@@ -0,0 +1,393 @@
+<template>
+  <div id="timeline_plot"></div>
+</template>
+<script>
+export default {
+  name: "TimelinePlot",
+  props: {
+    plotData: {
+      type: Object,
+      required: true,
+      default: () => { return null }
+    },
+  },
+  mounted() {
+    this.plot_timeline_graph();
+  },
+  methods: {
+    plot_timeline_graph() {
```

Comment:
This function is pretty complex.  Would it be reasonable to move some of the helper functions outside?  I'm a bit torn on this one, so if you think it's fine as-is, that's fine with me.  We can revisit some issues like this once we upgrade d3 as well.

Reply:
Yeah, I agree. Unfortunately due to the nature of d3's API, a lot of the objects that are returned from d3 functions (like `xAxis`) are used throughout the scope of `plot_timeline_graph()`. For this reason, I'm not sure if refactoring the code to different functions will result in cleaner code, but if you have suggestions or examples of other big-ish d3 plots that are better organized, I'd happily take a look.